### PR TITLE
Adding application API

### PR DESCRIPTION
**Pull request by brimoor**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -155,16 +150,126 @@ Wait until a job is complete and then download its output:
...
+usernames = api.list_users()
+```
+
+Create a new user:
+
+```py
+api.create_user("<username>")
+```
+
+And update the usage limits of a user:
```
Comment (hoddr) [inline README.md:230]: May be worth mentioning that the user limits are capped at the application's limits here

File: README.md [hunk 2]
```diff
@@ -155,16 +150,126 @@ Wait until a job is complete and then download its output:
...
+```
+
+### Performing User Actions
+
+To perform actions for a user of your application, you must first activate the
+user:
+
+```py
+# Activate an application user
+api.with_user("<username>")
```
Comment (hoddr) [inline README.md:249]: Is there an option to pass the desired user in like so?

```
api.list_data("<username>", ...params)

# or

api.list_data("<username>")(...params)
```

File: voxel51/api.py [hunk 3]
```diff
@@ -351,7 +364,7 @@ def query_jobs(self, jobs_query):
         return _parse_json_response(res)
 
     def upload_job_request(
-            self, job_request, job_name, auto_start=False, use_gpu=False,
+            self, job_request, job_name, auto_start=False, use_gpu=True,
```
Comment (hoddr) [inline voxel51/api.py:367]: I'm not sure defaulting to the more expensive option is best, but I understand that most people will likely want to use gpu for faster computation times. opt in versus opt out

File: voxel51/apps/auth.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        path (str): the path to an :class:`ApplicationToken` JSON file
+    '''
+    voxu.copy_file(path, TOKEN_PATH)
+    logger.info("ApplicationToken successfully activated")
+
+
+def deactivate_application_token():
+    '''Deactivates (deletes) the currently active application token, if any.
+
+    The active application token is at ``~/.voxel51/app-token.json``.
```
Comment (hoddr) [inline voxel51/apps/auth.py:41]: Looks like double ` marks here?

File: voxel51/apps/auth.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+def load_application_token(token_path=None):
+    '''Loads the active application token.
+
+    Args:
+        token_path: an optional path to an :class:`ApplicationToken` JSON file.
+            If no path is provided, the ``VOXEL51_APP_TOKEN`` environment
+            variable is checked and, if set, the token is loaded from that
+            path. Otherwise, the token is loaded from
+            ``~/.voxel51/app-token.json``
```
Comment (hoddr) [inline voxel51/apps/auth.py:59]: Or is the doubling of ` necessary?

File: README.md [hunk 6]
```diff
@@ -155,16 +150,126 @@ Wait until a job is complete and then download its output:
...
 ```py
-pprint(api.get_job_status(job_id))
+status = api.get_job_status(job_id)
+```
+
+
+## Application Quickstart
+
+This section provides a brief guide to using the Vision Services API with your
+application admin account.
```
Comment (hoddr) [inline README.md]: This relates to what we briefly touched upon Friday. I think we are being misleading by suggesting this is using the application admin's account. This is really a developer private key (token if you want) only for the application itself. The application admin's account is solely responsible for logging into the admin dashboard on the platform. Two linked but separate concepts.

**Review by hoddr — APPROVED** [review]
Overall LGTM

File: README.md (hunk 1 above)
Comment (brimoor) [reply README.md:230]: will address this in the platform docs.

File: README.md (hunk 6 above)
Comment (brimoor) [reply README.md]: Removed the notion of "app admin".

File: README.md (hunk 2 above)
Comment (brimoor) [reply README.md:249]: I thought about it but instead chose to make `ApplicationAPI` a subclass of `API` and just directly inherit all of the user methods. This way I didn't have to restub all of these methods for applications, repeat the docstrings, etc.

File: voxel51/api.py (hunk 3 above)
Comment (brimoor) [reply voxel51/api.py:367]: They will enjoy their jobs running faster :)

File: voxel51/apps/auth.py (hunk 4 above)
Comment (brimoor) [reply voxel51/apps/auth.py:41]: Required by the autodoc tool we're using here.

**Pull request merged**